go.mod

```diff
@@ -1,9 +1,9 @@
 module kpt.dev/configsync
 
-go 1.22
+go 1.22.4
```

Comment:
Is there a reason this pins to a patch version?

Reply:
It happens automatically now with Go 1.22 when you run `go mod tidy`. If I remove it, it comes back. Saw it happen in k8s upstream too.

Reply:
If I run `go mod tidy -go=1.22.0`, I can change it to  1.22.0, but If I try `go mod tidy -go=1.22` I get an error that `github.com/open-policy-agent/cert-controller@v0.10.2-0.20240531181455-2649f121ab97 requires go@1.22.0, but 1.22 is requested`

Reply:
Seems like if I run `go mod tidy -go=1.22.0` and then `go mod tidy`, it keeps the older version. So maybe that's the right move, to set the minimum requirement as low as possible.

Reply:
Does this dictate which Go patch version is used when binaries are built? If so, we might want this to be the latest patch version. The golang builder image is currently set to `1.22.5`

Reply:
AFAIK, the version in go.mod is just the minimum version requirement to build with.

Reply:
> The minimum version of Go required by the current module.

https://go.dev/doc/modules/gomod-ref

Reply:
For comparison, upstream k8s is also using 1.22.0: https://github.com/kubernetes/kubernetes/blob/master/go.mod#L9